ankihub/gui/utils.py

```diff
@@ -774,3 +775,37 @@ def get_ah_did_of_deck_or_ancestor_deck(anki_did: DeckId) -> Optional[uuid.UUID]
 
 def logged_into_ankiweb() -> bool:
     return bool(aqt.mw.pm.sync_auth())
+
+
+def refresh_anki_ui() -> None:
+    changes = OpChanges(
```

Comment:
I would simply call `mw.reset()` here to refresh everything so we don't have to worry about figuring which fields to set and hooks to call. Anki also calls `mw.update_undo_actions()`, which I think is needed because we clear the undo queue when we do direct DB write operations in some tasks.

Reply:
Hm, that might be better. However, when I replace `refresh_anki_ui()` with `aqt.mw.reset()`, Anki consistently lags:

https://github.com/user-attachments/assets/e6d87d6d-230f-4dcd-962e-16ea460b44ae

It doesn't happen with `refresh_anki_ui()`:

https://github.com/user-attachments/assets/a9122a35-b7cf-4b46-b7f9-ab83e38bf724

Reply:
It seems like the `update_undo_actions()` is causing the lag. I'll try to find out why

Reply:
Probably due to the backend call in _check_backend_undo_status()?

Reply:
Yes, _`backed.get_undo_status()` takes a long time:
![image](https://github.com/user-attachments/assets/478f08a4-eb93-4c88-be15-f36001453eb9)

Any idea why or how we can improve this?

Reply:
Oh, I think it might be because we are accessing the Anki DB to get review summaries/data at the same time

Reply:
Maybe moving the .run_in_background() calls after the refresh_anki_ui() call will help?

Reply:
Hm, but the lag still happens when the browser is open while you sync